include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl.h

```diff
@@ -1067,17 +1070,17 @@ struct __early_exit_find_or
             if (__shifted_idx < __n && __pred(__shifted_idx, __rngs...))
             {
                 if constexpr (_OrTagType::value)
-                {
                     __found_local.store(1);
-                    break;
-                }
                 else
                 {
                     for (auto __old = __found_local.load(); __comp(__shifted_idx, __old); __old = __found_local.load())
                     {
-                        __found_local.compare_exchange_strong(__old, __shifted_idx);
+                        if (__found_local.compare_exchange_strong(__old, __shifted_idx))
+                            break;
                     }
                 }
+
+                __something_was_found = true;
```

Comment:
Can we simplify this by using a break here?

Reply:
Sure, but my performance checks show me that this variant works a little bit faster than the variant with break.

Reply:
That's interesting. Does this performance benefit apply for the other break statements as well?

Reply:
I think required additional performance checks for the other cases.

Reply:
But this additional perf improvement looks too small, so rolled back.
Let's use `break` statement.